tests/backup/backup_helper.go

```diff
@@ -1629,6 +1667,310 @@ func restoreSuccessWithReplacePolicy(restoreName string, orgID string, retryDura
+		if forRestore {
+			// if we can transforming to obtain a restored specs, VolumeSnapshot should be ignored
+			if obj, ok := appSpecOrig.(*snapv1.VolumeSnapshot); ok {
+				log.Infof("TransformAppContextWithMappings is for restore contexts, ignoring transformation of 'VolumeSnapshot' [%s] in appContext [%s]", obj.Metadata.Name, appContextNamespace)
+				continue
+			} else if obj, ok := appSpecOrig.(*storageapi.StorageClass); ok {
+				log.Infof("TransformAppContextWithMappings is for restore contexts, ignoring transformation of 'StorageClass' [%s] in appContext [%s]", obj.Name, appContextNamespace)
+				continue
+			}
+		}
+
+		appSpec, err := CloneSpec(appSpecOrig) //clone spec to create "restore" specs
+		if err != nil {
+			err := fmt.Errorf("failed to clone spec: '%v'. Err: %v", appSpecOrig, err)
+			errors = append(errors, err)
+			continue
+		}
+		err = TransformToRestoredSpec(appSpec, storageClassMapping)
+		if err != nil {
+			err := fmt.Errorf("failed to TransformToRestoredSpec for %v, with sc map %s. Err: %v", appSpec, storageClassMapping, err)
+			errors = append(errors, err)
+			continue
+		}
+		err = UpdateNamespace(appSpec, namespaceMapping)
+		if err != nil {
+			err := fmt.Errorf("failed to Update the namespace for %v, with ns map %s. Err: %v", appSpec, namespaceMapping, err)
+			errors = append(errors, err)
+			continue
+		}
+		specObjects = append(specObjects, appSpec)
+
+		// TODO: remove workaround in future.
+		if specObj, ok := appSpecOrig.(*corev1.PersistentVolumeClaim); ok {
+			if _, ok := storageClassMapping[*specObj.Spec.StorageClassName]; !ok {
+				allStorageClassMappingsPresent = false
+			}
+		}
+	}
+
+	errStrings := make([]string, 0)
+	for _, err := range errors {
+		if err != nil {
+			errStrings = append(errStrings, err.Error())
+		}
+	}
+
+	if len(errStrings) > 0 {
+		return nil, fmt.Errorf("TransformAppContextWithMappings Errors: {%s}", strings.Join(errStrings, "}\n{"))
+	}
+
+	app := *appContext.App
+	app.SpecList = specObjects
+	restoreAppContext.App = &app
+
+	// we're having to do this as we're under the assumption that `ScheduleOptions.Namespace` will always contain the namespace of the scheduled app
+	options := CreateScheduleOptions("")
+	if namespace, ok := namespaceMapping[appContextNamespace]; ok {
+		options.Namespace = namespace
+	} else {
+		options.Namespace = appContextNamespace
+	}
+	restoreAppContext.ScheduleOptions = options
+
+	// TODO: remove workaround in future.
+	if !allStorageClassMappingsPresent {
+		restoreAppContext.SkipVolumeValidation = true
+	}
+
+	return &restoreAppContext, nil
+}
+
+// CloneSpec clones a given spec and returns it. It returns an error if the object (spec) provided is not supported by this function
+func TransformToRestoredSpec(spec interface{}, storageClassMapping map[string]string) error {
+	if specObj, ok := spec.(*corev1.PersistentVolumeClaim); ok {
+		if sc, ok := storageClassMapping[*specObj.Spec.StorageClassName]; ok {
```

Comment:
How about returning an error if it's not okay? Also, log specObj.Spec.StorageClassName, sc.

Reply:
it's not an error if it's not okay. This is just transforming stuff _if_ it exists.
I feel that there's no point in logging this as this is fed in from the outside. Whatever logging needs to be done, needs to be done by the person who created the mapping as they own the data, not I. Though, logging _if_ there is PVC in itself is useful as that also tells us that SCMapping took place, though that requires unnecessary logic.

Reply:
The function signature implies that it should throw an error, but the function does not actually throw any errors.

Reply:
changed it. see latest commit